sarif/sarif.go

```diff
@@ -0,0 +1,578 @@
+package sarif
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/go-errors/errors"
+	"github.com/jonboulle/clockwork"
+	"github.com/package-url/packageurl-go"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
```

Comment:
Extra blank line here

Reply:
if this is the only review finding, i may have finally learned how to code, excuse me while i throw a mini party in the corner
https://www.youtube.com/watch?v=QSZM0khJFdM